chia/cmds/coin_funcs.py

```diff
@@ -0,0 +1,163 @@
+    print(f"{len(conf_coins)} confirmed coins.")
+    print(f"{len(unconfirmed_additions)} unconfirmed additions.")
+    print(f"{len(unconfirmed_removals)} unconfirmed removals.")
+    print("Confirmed coins:")
+    for cr in conf_coins:
+        address = encode_puzzle_hash(cr.coin.puzzle_hash, addr_prefix)
+        amount = print_balance(cr.coin.amount, mojo_per_unit, addr_prefix)
+        print(f"Coin ID: 0x{cr.name.hex()}")
+        print(f"    Current Address: {address} Amount: {amount}, Confirmed in block: {cr.confirmed_block_index}\n")
+    if show_unconfirmed:
+        print("\nUnconfirmed Removals:")
+        for cr in unconfirmed_removals:
+            address = encode_puzzle_hash(cr.coin.puzzle_hash, addr_prefix)
+            amount = print_balance(cr.coin.amount, mojo_per_unit, addr_prefix)
+            print(f"Coin ID: 0x{cr.name.hex()}")
+            print(f"    Previous Address: {address} Amount: {amount}, Confirmed in block: {cr.confirmed_block_index}")
+        print("\nUnconfirmed Additions:")
+        for coin in unconfirmed_additions:
+            address = encode_puzzle_hash(coin.puzzle_hash, addr_prefix)
+            amount = print_balance(coin.amount, mojo_per_unit, addr_prefix)
+            print(f"Coin ID: 0x{coin.name().hex()}")
+            print(f"    New Address: {address} Amount: {amount}, Not yet confirmed in a block.\n")
+
+
+async def async_combine(args: Dict[str, Any], wallet_client: WalletRpcClient, fingerprint: int) -> None:
+    wallet_id: int = args["id"]
+    min_coin_amount = Decimal(args["min_coin_amount"])
+    excluded_amounts = args["excluded_amounts"]
+    number_of_coins = args["number_of_coins"]
+    max_dust_amount = Decimal(args["max_dust_amount"])
+    fee = Decimal(args["fee"])
+    if number_of_coins > 500:
+        raise ValueError(f"{number_of_coins} coins is greater then the maximum limit of 500 coins.")
+    try:
+        wallet_type = await get_wallet_type(wallet_id=wallet_id, wallet_client=wallet_client)
+        mojo_per_unit = get_mojo_per_unit(wallet_type)
+    except LookupError:
+        print(f"Wallet id: {wallet_id} not found.")
+        return
+    if not await wallet_client.get_synced():
+        print("Wallet not synced. Please wait.")
+        return
+    final_max_dust_amount = uint64(int(max_dust_amount * mojo_per_unit))
+    final_min_coin_amount: uint64 = uint64(int(min_coin_amount * mojo_per_unit))
+    final_excluded_amounts: List[uint64] = [uint64(int(amount * mojo_per_unit)) for amount in excluded_amounts]
+    final_fee = uint64(int(fee * mojo_per_unit))
+    conf_coins, _, _ = await wallet_client.get_spendable_coins(
+        wallet_id=wallet_id,
+        max_coin_amount=final_max_dust_amount,
+        min_coin_amount=final_min_coin_amount,
+        excluded_amounts=final_excluded_amounts,
+    )
+    if len(conf_coins) <= 1:
+        print("No coins to combine.")
+        return
+    conf_coins.sort(key=lambda r: r.coin.amount)  # sort the smallest first
```

Comment:
I'm not exactly sure how to go about it, but I feel like there should be more granular control over which coins you're combining.  This is a good default, but you may just be looking to get a coin of a certain amount for example, or you may specifically want to combine a few specific coins.